types/src/v0/impls/header.rs

```diff
@@ -747,13 +747,11 @@ impl BlockHeader<SeqTypes> for Header {
         let mut validated_state = parent_state.clone();
 
         let chain_config = if version > instance_state.current_version {
-            match instance_state
-                .upgrades
-                .get(&version)
-                .map(|upgrade| match upgrade.upgrade_type {
+            match instance_state.upgrades.get(&version) {
+                Some(upgrade) => match upgrade.upgrade_type {
                     UpgradeType::ChainConfig { chain_config } => chain_config,
```

Comment:
Not looking for a change here,  just trying to improve my understanding. These semantics are confusing to me. Why do we have an `UpgradeType:ChainConfig`. Couldn't we just be upgrading between versions, and have the version define what is included in the upgrade? For example, I would imaging that the upgrade type would be `Marketplace` and `ChainConfig` upgrade would happen "under the hood" since it is a dependency of marketplace.

Reply:
hmm I think renaming this upgrade type to FeeUpgrade would be better. So, fee upgrade would require a new chain config that is provided in the toml file.

Reply:
Ah! yes I think so too.